### PR TITLE
Data 377

### PR DESCRIPTION
**Pull request by Santhin**
Added possibility to run a script inside docker image when `execution_script` is enabled and when datahub url is provided it's gonna be ingested to env var inside `KubernetesPodOperator`

Resolves `DATA-377`

---
Keep in mind:
- [ ] Documentation updates
- [ ] [Changelog](CHANGELOG.md) updates

### CODE REVIEW

File: dbt_airflow_factory/k8s/k8s_parameters.py [hunk 1]
```diff
@@ -56,6 +56,7 @@ def __init__(
         secrets: Optional[List[Secret]] = None,
         is_delete_operator_pod: bool = True,
         config_file: str = "~/.kube/config",
+        execution_script: bool = False,
```
Comment (p-pekala) [inline dbt_airflow_factory/k8s/k8s_parameters.py:59]: that should be script name, not boolean. We want user to define their own scripts if needed.

File: dbt_airflow_factory/k8s/k8s_operator.py [hunk 2]
```diff
@@ -52,10 +52,15 @@ def create(
...
+        command: str,
+        model: Optional[str],
+        additional_dbt_args: Optional[List[str]],
     ) -> List[str]:
         args = [
-            "set -e;",
+            "./cmd_executor.sh"
+            if self.kubernetes_execution_parameters.execution_script
+            else "set -e;",
             f"dbt --no-write-json {command}",
```
Comment (p-pekala) [inline dbt_airflow_factory/k8s/k8s_operator.py:64]: we dicussed this "dbt --no-write-json" should go into else

File: dbt_airflow_factory/tasks_builder/builder.py [hunk 3]
```diff
@@ -140,9 +142,11 @@ def _create_tasks_from_graph(self, dbt_airflow_graph: DbtAirflowGraph) -> ModelE
         }
         for node, neighbour in dbt_airflow_graph.graph.edges():
             # noinspection PyStatementEffect
-            result_tasks[node].get_end_task() >> result_tasks[neighbour].get_start_task()
+            (result_tasks[node].get_end_task() >> result_tasks[neighbour].get_start_task())
```
Comment (p-pekala) [inline dbt_airflow_factory/tasks_builder/builder.py:145]: why extra braket?

File: tests/test_config_propagation.py [hunk 4]
```diff
@@ -41,12 +40,47 @@ def test_configuration():
...
+        == 'set -e; dbt --no-write-json run --target dev --vars "{}" --project-dir /dbt --profiles-dir /root/.dbt --select dim_users'  # noqa: E501
+    )
     assert run_task.config_file == "/usr/local/airflow/dags/kube_config.yaml"
     assert run_task.is_delete_operator_pod
     assert "--project-dir /dbt" in run_task.arguments[0]
     assert "--profiles-dir /root/.dbt" in run_task.arguments[0]
     assert "--target dev" in run_task.arguments[0]
+
+
+def test_configuration_with_qa_config():
```
Comment (p-pekala) [inline tests/test_config_propagation.py:68]: I feel like it should be a separate file for qa testing. It is not only configuration testing.

File: dbt_airflow_factory/k8s/k8s_operator.py [hunk 5]
```diff
@@ -52,10 +52,15 @@ def create(
...
     def _prepare_arguments(
-        self, command: str, model: Optional[str], additional_dbt_args: Optional[List[str]]
+        self,
+        command: str,
+        model: Optional[str],
+        additional_dbt_args: Optional[List[str]],
     ) -> List[str]:
         args = [
-            "set -e;",
+            "./cmd_executor.sh"
```
Comment (p-pekala) [inline dbt_airflow_factory/k8s/k8s_operator.py:61]: script path/name should be parameterized

**Review by p-pekala — CHANGES_REQUESTED** [review]

File: dbt_airflow_factory/tasks_builder/builder.py (hunk 3 above)
Comment (Santhin) [reply dbt_airflow_factory/tasks_builder/builder.py:145]: Automatically formated by pre-commit

File: dbt_airflow_factory/k8s/k8s_parameters.py (hunk 1 above)
Comment (Santhin) [reply dbt_airflow_factory/k8s/k8s_parameters.py]: Changed: 348912fe75fb1ff0213845d4de02a3d697a55bc2

File: dbt_airflow_factory/k8s/k8s_operator.py (hunk 2 above)
Comment (Santhin) [reply dbt_airflow_factory/k8s/k8s_operator.py]: Changed: 86f643c19307ffd429c599177f3396e4eb186920

**Review by p-pekala — APPROVED** [review]

**Pull request merged**